Google.PowerShell/BigQuery/BqJob.cs

```diff
@@ -178,5 +178,251 @@ public void DoGetRequest()
+        [ValidateNotNullOrEmpty]
+        public string QueryString { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// Specifies BigQuery's legacy SQL dialect for this query.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.DoQuery)]
+        public SwitchParameter UseLegacySql { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The dataset to use for any unqualified table names in QueryString.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.DoQuery)]
+        [PropertyByTypeTransformation(TypeToTransform = typeof(Dataset),
+            Property = nameof(Dataset.DatasetReference))]
+        [Alias("Dataset")]
+        public DatasetReference DefaultDataset { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The destination table to write the results into. If this is not specified, the 
+        /// results will be stored in a temporary table.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.DoQuery)]
+        [PropertyByTypeTransformation(TypeToTransform = typeof(Table),
+            Property = nameof(Table.TableReference))]
+        [Alias("Table", "Dest")]
+        public TableReference DestinationTable { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// Priority of the query.  Can be 'Batch' or 'Interactive'.
+        /// </para>
+        /// </summary>
+        public QueryPriority Priority { get; set; }
+
+        // Copy Parameters.
+
+        /// <summary>
+        /// <para type="description">
+        /// Selects job type Copy.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, ParameterSetName = ParameterSetNames.DoCopy)]
+        public SwitchParameter Copy { get; set; }
+
+        // Load Parameters.
+
+        /// <summary>
+        /// <para type="description">
+        /// Selects job type Load.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, ParameterSetName = ParameterSetNames.DoLoad)]
+        public SwitchParameter Load { get; set; }
+
+        // Extract Parameters.
+
+        /// <summary>
+        /// <para type="description">
+        /// Selects job type Extract.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, ParameterSetName = ParameterSetNames.DoExtract)]
+        public SwitchParameter Extract { get; set; }
+
+        protected override void ProcessRecord()
+        {
+            Apis.Bigquery.v2.Data.Job result;
+
+            switch (ParameterSetName)
+            {
+                case ParameterSetNames.DoQuery:
+                    result = DoQuery(Client);
```

Comment:
Part of the point of doing the Client property was you didn't have to pass this parameter anymore.